src/player.cpp

```diff
@@ -3751,6 +3751,18 @@ void Player::addAttacked(const Player* attacked)
 	attackedSet.insert(attacked->guid);
 }
 
+void Player::removeAttacked(const Player* attacked)
+{
+	if (!attacked || attacked == this) {
+		return;
+	}
+
+	auto it = attackedSet.find(attacked->guid);
```

Comment:
@gugahoa he doesn't need to check if `hasAttacked` prior to calling `removeAttacked`, not the other way around